app/controllers/publish/courses/study_mode_controller.rb

```diff
@@ -43,6 +85,10 @@ def current_step
       def errors
         params.dig(:course, :study_mode) ? {} : { study_mode: [I18n.t('activemodel.errors.models.publish/course_study_mode_form.attributes.study_mode.blank')] }
       end
+
+      def previous_tda_course_path?
+        params[:previous_tda_course] == 'true'
+      end
     end
   end
 end
```

Comment:
It is worth a try to move the controller logic to the basic detail concern to have less source of truth for the add course workflow BUT do this as last thing and I suggest to timebox to a hour maximum. @dcyoung-dev wdyt?

Reply:
Can we do this as part of another card? Scope creep is real on this card and the PR is big enough now. 

I think we can have a card just to refactor.